### PR TITLE
Cache screen CTM

### LINKED ISSUE #436: perf improvement: investigate caching matrix from getScreenCTM
Mouse down events end up calling getScreenCTM when trying to save the coordinates of at the start of a drag.  This causes a style recalculation and a forced layout (see screenshot of timeline).  Similarly to #435, I don't think the matrix should change unless workspace resize ends  up being called.  Look into caching it.

screenctm

The delay when starting a drag is most visible on mobile.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
Fixes #436

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
@picklesrus @NeilFraser 
Please try this out and try to break it.

File: core/utils.js [hunk 1]
```diff
@@ -308,14 +308,13 @@ Blockly.isRightButton = function(e) {
  * The origin (0,0) is the top-left corner of the Blockly svg.
  * @param {!Event} e Mouse event.
  * @param {!Element} svg SVG element.
+ * @param {SVGMatrix} matrix Inverted screen CTM to use.
```
Comment (rachel-fenichel) [inline core/utils.js]: One concern here is that the matrix could be null (it starts out null in workspace_svg).  I suppose in that case I should fall back to getting the matrix directly from the svg.

**Comment by picklesrus** [comment]
FWIW, I grabbed a copy and played around a bunch, and tested all the things I broke with my resize change since they hit similar branches of code.  No issues found so far.

**Comment by NeilFraser** [comment]
LGTM

**Pull request merged**